braindecode/models/deep4.py

```diff
@@ -270,17 +278,20 @@ def add_conv_pool_block(
         self.eval()
         if self.final_conv_length == "auto":
             self.final_conv_length = self.get_output_shape()[2]
-        self.add_module(
-            "conv_classifier",
-            nn.Conv2d(
-                self.n_filters_4,
-                self.n_outputs,
-                (self.final_conv_length, 1),
-                bias=True,
-            ),
-        )
-        self.add_module("softmax", nn.LogSoftmax(dim=1))
-        self.add_module("squeeze", Expression(squeeze_final_output))
+
+        # The conv_classifier will be the final_layer and the other ones will be incorporated
+        module = nn.Sequential()
+
+        module.add_module("conv_classifier",
+                          nn.Conv2d(n_filters_4, self.n_outputs,
+                                    (self.final_conv_length, 1), bias=True, ))
+
+        module.add_module("softmax", nn.LogSoftmax(dim=1))
+
+        module.add_module("squeeze", Expression(squeeze_final_output))
+
+        # The conv_classifier will be the final_layer and the other ones will be incorporated
+        self.add_module("final_layer", module)
```

Comment:
For this part i would use the code snippet provided by @PierreGtch https://github.com/braindecode/braindecode/pull/520#issuecomment-1713563121